src/net/tcp/stream.rs

```diff
@@ -103,6 +103,22 @@ impl TcpStream {
     pub async fn write<T: BoundedBuf>(&self, buf: T) -> crate::BufResult<usize, T> {
         self.inner.write(buf).await
     }
+    /// Write some data to the stream from the buffer. Will attempt to do so without intermediate copies.
+    ///
+    /// On success, returns the number of bytes written.
+    ///
+    /// See the linux [kernel docs](https://www.kernel.org/doc/html/latest/networking/msg_zerocopy.html)
+    /// for a discussion on when this might be appropriate. In particular:
+    ///
+    /// > Copy avoidance is not a free lunch. As implemented, with page pinning,
+    /// > it replaces per byte copy cost with page accounting and completion
+    /// > notification overhead. As a result, zero copy is generally only effective
+    /// > at writes over around 10 KB.
+    ///
+    /// Note: Using fixed buffers [#54](https://github.com/tokio-rs/tokio-uring/pull/54), avoids the page-pinning overhead
+    pub async fn write_zc<T: BoundedBuf>(&self, buf: T) -> crate::BufResult<usize, T> {
```

Comment:
What's the purpose of calling this `write_zc`? Where does that come from? Does the std or tokio libraries have a name like that for a TCP connection? If not, at first glance it seems better to  stick with the names of either the io_uring interface or maybe the io-uring crate API.

Reply:
`write_zc` is zero copy version of existing `write`, similar to how  `UdpSocket::send_zc` is zero copy version of `UdpSocket::send`.

Reply:
Tokio doesn't have this, but Linux sockets do, and it's a very useful optimisation. Offloading data copy from your CPU to your NIC can vastly reduce CPU overhead